features/emojiMod.js

```diff
@@ -129,36 +133,147 @@ const reactionHandlers = {
+    if (!isStaff(member)) {
+      return;
+    }
+
+    const restrictedRole = message.guild.roles.find(
+      role => role.name === "restricted"
+    );
+    const targetUser = message.guild.members.get(message.author.id);
+
+    const modLogChannel = bot.channels.find(
+      channel => channel.name === "mod-log"
+    );
+
+    const startDate = new Date();
+    const endDate = new Date();
+    endDate.setDate(startDate.getDate() + config.RESTRICTION_DAYS);
+
+    let logMessage = `<@${message.author.id}> has been restricted for ${config.RESTRICTION_DAYS} days by <@${member.user.id}>.`;
+    logMessage += "\n\n";
+    logMessage += "```";
+    logMessage += `Start Date: ${startDate.toLocaleString()}`;
+    logMessage += "\n";
+    logMessage += `End Date: ${endDate.toLocaleString()}`;
+    logMessage += "```";
+
+    targetUser.roles.add(restrictedRole);
+    modLogChannel.send(logMessage);
+
+    knex
+      .table("restrictions")
+      .insert({
+        user_id: message.author.id,
+        start_time: startDate,
+        end_time: endDate
+      })
+      .then(() => {
+        console.log(`Saved restriction for ${message.author.id}`);
+      });
   }
 };
 
-const emojiMod = bot => ({
-  handleReaction: ({ reaction, user }) => {
-    const { message } = reaction;
-    const { member } = message;
-    if (user.id === bot.user.id) {
-      return;
-    }
-    if (message.author.id === bot.user.id) {
+const setupRestrictionsIntervals = async bot => {
+  const expiredRestrictions = await knex("restrictions").where(
+    "end_time",
+    "<",
+    new Date()
+  );
+
+  const guild = bot.guilds.find(g => g.name === "Reactiflux");
+
+  expiredRestrictions.forEach(expiredRestriction => {
+    const { id, user_id, start_time } = expiredRestriction;
+    const startDate = new Date(start_time).toLocaleString();
+
+    const modLogChannel = bot.channels.find(
+      channel => channel.name === "mod-log"
+    );
+
+    let logMessage = `<@${user_id}>'s restriction from ${startDate} has been lifted.`;
+    modLogChannel.send(logMessage);
+
+    const restrictedRole = guild.roles.find(role => role.name === "restricted");
+    const targetUser = guild.members.get(user_id);
+
+    knex("restrictions")
+      .where({ user_id })
+      .delete();
+
+    if (!targetUser) {
+      // on the wrong server, not a problem
       return;
     }
-    let emoji = reaction.emoji.toString();
 
-    if (thumbsDownEmojis.includes(emoji)) {
-      emoji = "👎";
-    }
+    targetUser.roles.remove(restrictedRole);
+  });
+};
 
-    const reactionHandler = reactionHandlers[emoji];
-    if (reactionHandler) {
-      reactionHandler(
-        bot,
-        reaction,
-        message,
-        message.guild.members.get(user.id)
+const emojiMod = bot => {
+  setTimeout(() => {
+    setupRestrictionsIntervals(bot);
+
+    setInterval(() => {
+      setupRestrictionsIntervals(bot);
+    }, 12 * 60 * 60 * 1000); // 12 hours
+  }, 10000);
```

Comment:
How come this is done in a `setTimeout`? 🤔

Reply:
@thisRaptori *sigh*

The `bot` isn't fully done setting up at that point, so it can't read members/servers immediately. It's _almost_ done setting up, but not quite, so there's the timeout. I tried like ~3-4 seconds but it wasn't 100% when joining RF.

If you want to (:eyes:) you have to hook it up to the lifecycle and initialize it in there, but afaict that would be a pretty decent restructure for this file that I deemed wasn't worth it.

Reply:
Yeah makes sense, no point completely redoing it all just for this 👍